### DISCUSSION TITLE
Add a low / configurable chance of generating a new bounty for a board that just became 'rendered' and is empty

### DISCUSSION THREAD

**Discussion by Armadelix** [ejektaflex/Bountiful#248] [Mod Ideas]
Hello,

Very thankful for the existence of this mod.
I personally tweaked the config to generate new bounties every 10'ish minutes as I don't want them to generate so often, the problem that seems to surface because of that is that bounty boards remain empty in chunks that are not active for that long, which is quite a bummer for whoever has the preference of less frequent generation of bounties.

As the title reads, I would like to suggest implementing a chance modifier (hopefully configurable) for the bounty board that will trigger a generation of a bounty; assuming the board is empty and has just became active / chunk became active. 
Hopefully, it can remedy the supposed problem of boards remaining empty due to low frequency of bounties generation in chunks that are not active for long - without being taxing on performance too(I think).

Thanks in advance.

**Comment by ejektaflex** [comment]
Hi! So, it sounds like the problem is that when a chunk is unloaded, bounties don't generate inside of it. Because of that, taking a trip to a village might not yield many new bounties, especially with a long generation interval, right?

We could definitely do a check to see when the last bounty was generated, and "catch up" the board if it has been a while. E.g. if the board hasn't generated a bounty in the last 30 minutes (1800 seconds) because the chunk was unloaded, and bounties generate every 90 seconds,  then we ought to add 20 (1800/90) bounties to the board.

This would probably cap at the number of bounty slots (24). In your situation, this would mean that if they visited 20 minutes later, two new bounties would generate.

This would solve a longstanding issue of visiting villages and not seeing much new content :)

I'll look into implementing this!

**Reply by Armadelix** [reply]
Thanks for the elaborate reply! This is so nice to read.
